code/clue.js

```diff
@@ -1,59 +1,396 @@
+  id: 'poison',
+  image: 'assets/poison.jpg'
+};
+
+const axe = {
+  name: 'an axe',
+  weight: 12,
+  flexibility: 3,
+  color: '#7b497f',
+  id: 'axe',
+  image: 'assets/axe.jpg'
+};
+
+const bat = {
+  name: 'a bat',
+  weight: 3,
+  flexibility: 7,
+  color: '#abb782',
+  id: 'bat',
+  image: 'assets/bat.jpg'
+};
+
+const trophy = {
+  name: 'a trophy',
+  weight: 5,
+  flexibility: 3,
+  color: '#fce853',
+  id: 'trophy',
+  image: 'assets/trophy1.jpg'
+};
+
+const pistol = {
+  name: ' a pistol',
+  weight: 1,
+  flexibility: 9,
+  color: '#d1a93c',
+  id: 'pistol',
+  image: 'assets/pistol.jpg'
+};
+
+// OBJECTS FOR THE ROOMS (BECAUSE THEY HAVE PICTURES)
+const diningRoom = {
+  name: 'Dining Room',
+  image: 'assets/dining.jpg'
+};
+
+const conservatory = {
+  name: 'Conservatory',
+  image: 'assets/conservatory.jpg'
+};
+
+const kitchen = {
+  name: 'Kitchen',
+  image: 'assets/kitchen.jpg'
+};
+
+const study = {
+  name: 'Study',
+  image: 'assets/study.jpg'
+};
 
-// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
+const library = {
+  name: 'Library',
+  image: 'assets/library.jpg'
+};
 
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
+const billiardRoom = {
+  name: 'Billiard Room',
+  image: 'assets/billiard.jpg'
+};
 
+const lounge = {
+  name: 'Lounge',
+  image: 'assets/lounge.jpg'
+};
+
+const ballRoom = {
+  name: 'Ball Room',
+  image: 'assets/ballroom.jpg'
+};
+
+const hall = {
+  name: 'Hall',
+  image: 'assets/hall.jpg'
+};
+
+const spa = {
+  name: 'Spa',
+  image: 'assets/spa.jpg'
+};
+
+const livingRoom = {
+  name: 'Living Room',
+  image: 'assets/livingroom.jpg'
+};
+
+const observatory = {
+  name: 'Observatory',
+  image: 'assets/observatory.jpg'
+};
+
+const theater = {
+  name: 'Theater',
+  image: 'assets/theater.jpg'
+};
+
+const guestHouse = {
+  name: 'Guest House',
+  image: 'assets/guesthouse.jpg'
+};
+
+const patio = {
+  name: 'Patio',
+  image: 'assets/patio.jpg'
+};
+
+// STEP 2. NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
 const suspects = [
   mrGreen,
+  professorPlum,
+  missScarlet,
+  mrsPeacock,
+  colonelMustard,
   mrsWhite
-  // ...  and the rest
-]
+];
 
-const weapons = []
+const weapons = [
+  rope,
+  knife,
+  candleStick,
+  dumbBell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
+];
 
-const rooms = []
+const rooms = [
+  diningRoom,
+  conservatory,
+  kitchen,
+  study,
+  library,
+  billiardRoom,
+  lounge,
+  ballRoom,
+  hall,
+  spa,
+  livingRoom,
+  observatory,
+  theater,
+  guestHouse,
+  patio
+];
 
 // THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
 // YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
+const randomSelector = array => { //I don't understand this
+  return array[Math.floor(Math.random() * array.length)] //What does array.length do here?
```

Comment:
This code generates a random index number. And array.length ensures that the number is relevant to the array in question. Math.random generates a random number between 0 - 1. If we were to multiply this by 10 then we would get a random number between 1 - 10. But by using this code it will multiply that random number by the length of the array. And means that the index number will be between 0 - 11 if the array in question has a length of 12. 
I hope this helps?

Reply:
Thank you, very clear explanation Jamie 👍